app/assets/javascripts/devices/devices-show.es6

```diff
@@ -0,0 +1,182 @@
+$(document).on('page:change', function() {
+  if ($(".c-friends.a-show_device").length === 1 || $(".c-devices.a-show").length === 1) {
```

Comment:
I'd suggest to move checks like `$(".c-friends.a-show_device").length === 1` info separate function and give that function a name that could tell us, what we really want to check here.